src/XMLConverter.php

```diff
@@ -80,27 +80,45 @@ class XMLConverter
     ];
 
     /**
-     * Convert an Record collection into a DOMDocument.
+     * Convert a Record collection into a DOMDocument.
      *
      * @param array|Traversable $records the CSV records collection
+     *
+     * @return DOMDocument
      */
     public function convert($records): DOMDocument
+    {
+        $doc = new DOMDocument('1.0');
+        $doc->appendChild(
+            $this->import($records, $doc)
+        );
+        return $doc;
+    }
+
+    /**
+     * Create a new DOMElement related to the given DOMDocument.
+     *
+     * **DOES NOT** attach to the DOMDocument
+     *
+     * @param iterable $records
+     *
+     * @return DOMElement
+     */
+    public function import($records, DOMDocument $doc): DOMElement
     {
         if (!is_iterable($records)) {
```

Comment:
I'm wondering if we should not duplicate this check in the `XMLConverter:convert` method otherwise we are introducing a subtle BC break when throwing exception 🤔  With the current implementation the exception will always comes from the import which is true but not expected from a BC point of view .

Reply:
Agree. Added to both methods.